client/src/client.rs

```diff
@@ -635,14 +657,44 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		Ok(proof)
 	}
 
-	/// Returns changes trie configuration and storage or an error if it is not supported.
-	fn require_changes_trie(&self) -> sp_blockchain::Result<(ChangesTrieConfiguration, &B::ChangesTrieStorage)> {
-		let config = self.changes_trie_config()?;
-		let storage = self.backend.changes_trie_storage();
-		match (config, storage) {
-			(Some(config), Some(storage)) => Ok((config, storage)),
-			_ => Err(sp_blockchain::Error::ChangesTriesNotSupported.into()),
+	/// Returns changes trie storage and all configurations that have been active in the range [first; last].
+	///
+	/// Configurations are returned in descending order (and obviously never overlap).
+	/// If prefer_configs is true, returns maximal consequent configurations ranges, starting from last and
+	/// stopping on either first, or when CT have been disabled.
+	/// Fails if or an error if it is not supported.
+	fn require_changes_trie(
+		&self,
+		first: NumberFor<Block>,
+		last: Block::Hash,
+		prefer_configs: bool,
+	) -> sp_blockchain::Result<(
+		&dyn PrunableStateChangesTrieStorage<Block, Blake2Hasher>,
+		Vec<(NumberFor<Block>, Option<(NumberFor<Block>, Block::Hash)>, ChangesTrieConfiguration)>,
```

Comment:
I think I have seen a similar tuple already, could maybe change to a struct with named fields.

Reply:
It maybe seems similar, but it is not the same. I'll think of replacing some tuples with structs, where appropriate.